pkg/scheduler/scheduler.go

```diff
@@ -305,45 +307,78 @@ func New(client clientset.Interface,
-		profiles:                     append([]schedulerapi.KubeSchedulerProfile(nil), options.profiles...),
-		registry:                     registry,
-		nodeInfoSnapshot:             snapshot,
-		extenders:                    options.extenders,
-		frameworkCapturer:            options.frameworkCapturer,
-		parallellism:                 options.parallelism,
-		clusterEventMap:              clusterEventMap,
+	extenders, err := buildExtenders(options.extenders, options.profiles)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't build extenders: %w", err)
 	}
 
-	metrics.Register()
+	// The nominator will be passed all the way to framework instantiation.
+	nominator := internalqueue.NewPodNominator(informerFactory.Core().V1().Pods().Lister())
+	snapshot := internalcache.NewEmptySnapshot()
+	clusterEventMap := make(map[framework.ClusterEvent]sets.String)
 
-	// Create the config from component config
-	sched, err := configurator.create()
+	profiles, err := profile.NewMap(options.profiles, registry, recorderFactory,
+		frameworkruntime.WithComponentConfigVersion(options.componentConfigVersion),
+		frameworkruntime.WithClientSet(client),
+		frameworkruntime.WithKubeConfig(options.kubeConfig),
+		frameworkruntime.WithInformerFactory(informerFactory),
+		frameworkruntime.WithSnapshotSharedLister(snapshot),
+		frameworkruntime.WithPodNominator(nominator),
+		frameworkruntime.WithCaptureProfile(frameworkruntime.CaptureProfile(options.frameworkCapturer)),
+		frameworkruntime.WithClusterEventMap(clusterEventMap),
+		frameworkruntime.WithParallelism(int(options.parallelism)),
+		frameworkruntime.WithExtenders(extenders),
+	)
 	if err != nil {
-		return nil, fmt.Errorf("couldn't create scheduler: %v", err)
+		return nil, fmt.Errorf("initializing profiles: %v", err)
+	}
+
+	if len(profiles) == 0 {
+		return nil, errors.New("at least one profile is required")
 	}
 
+	podQueue := internalqueue.NewSchedulingQueue(
+		profiles[options.profiles[0].SchedulerName].QueueSortFunc(),
+		informerFactory,
+		internalqueue.WithPodInitialBackoffDuration(time.Duration(options.podInitialBackoffSeconds)*time.Second),
+		internalqueue.WithPodMaxBackoffDuration(time.Duration(options.podMaxBackoffSeconds)*time.Second),
+		internalqueue.WithPodNominator(nominator),
+		internalqueue.WithClusterEventMap(clusterEventMap),
+		internalqueue.WithPodMaxUnschedulableQDuration(options.podMaxUnschedulableQDuration),
+	)
+
+	schedulerCache := internalcache.New(durationToExpireAssumedPod, stopEverything)
+
+	// Setup cache debugger.
+	debugger := cachedebugger.New(
+		informerFactory.Core().V1().Nodes().Lister(),
+		informerFactory.Core().V1().Pods().Lister(),
+		schedulerCache,
+		podQueue,
+	)
+	debugger.ListenForSignal(stopEverything)
+
+	sched := newScheduler(
+		schedulerCache,
+		extenders,
+		internalqueue.MakeNextPodFunc(podQueue),
+		MakeDefaultErrorFunc(client, informerFactory.Core().V1().Pods().Lister(), podQueue, schedulerCache),
```

Comment:
We can define a variable for `informerFactory.Core().V1().Pods().Lister()` as it's long and used several times in this function after this reactor. :)

Reply:
Marked it down. I'll do this in another follow-up PR. Thanks.

Reply:
Not need to do it in other pr or commit. This is introduced by this refactoring.

Reply:
I haven't checked it yet. Let me see.

Reply:
done :)